pinot-spi/src/main/java/org/apache/pinot/spi/env/PinotConfiguration.java

```diff
@@ -170,8 +174,18 @@ public static List<Configuration> applyDynamicEnvConfig(List<Configuration> conf
       Map<String, String> environmentVariables) {
     return configurations.stream().peek(configuration -> {
       for (String dynamicEnvConfigVarName : configuration.getStringArray(ENV_DYNAMIC_CONFIG_KEY)) {
-        configuration.setProperty(dynamicEnvConfigVarName,
-            environmentVariables.get(configuration.getString(dynamicEnvConfigVarName)));
+        if (configuration.getProperty(TEMPLATED_KEY) == null || !configuration.getProperty(TEMPLATED_KEY)
+            .equals(true)) {
```

Comment:
```suggestion
        if (configuration.getBoolean(TEMPLATED_KEY, false)) {
```

Reply:
This check can be moved before the for loop

Reply:
Completely forgot about this function. Thanks!